crates/iceberg/src/transaction.rs

```diff
@@ -133,10 +134,15 @@ impl<'a> Transaction<'a> {
     /// Creates a fast append action.
     pub fn fast_append(
         self,
+        snapshot_id: Option<i64>,
         commit_uuid: Option<Uuid>,
         key_metadata: Vec<u8>,
     ) -> Result<FastAppendAction<'a>> {
-        let snapshot_id = self.generate_unique_snapshot_id();
+        let snapshot_id = if let Some(snapshot_id) = snapshot_id {
+            snapshot_id
```

Comment:
We should ensure the snapshot id is unique in snapshots. Otherwise we should return the error and let client to regenerate it.
```    
fn generate_unique_snapshot_id(&self) -> i64 {
       ...
        let mut snapshot_id = generate_random_id();
        while self
            .table
            .metadata()
            .snapshots()
            .any(|s| s.snapshot_id() == snapshot_id)
        {
            snapshot_id = generate_random_id();
        }
        snapshot_id
    }
```